hazelcast/src/test/java/com/hazelcast/cache/merge/CacheSplitBrainTest.java

```diff
@@ -0,0 +1,380 @@
+
+        Cache cache1 = cacheManager1.createCache(cacheName, cacheConfig);
+        Cache cache2 = cacheManager2.createCache(cacheName, cacheConfig);
+
+        // TODO We assume that until here and also while doing get/put, cluster is still splitted.
+        // This assumptions seems fragile due to time sensitivity.
+
+        cache1.put("key1", "higherHitsValue");
+        cache1.put("key2", "value2");
+
+        // Increase hits number
+        cache1.get("key1");
+        cache1.get("key1");
+
+        cache2.put("key1", "value1");
+        cache2.put("key2", "higherHitsValue2");
+
+        // Increase hits number
+        cache2.get("key2");
+        cache2.get("key2");
+
+        assertOpenEventually(lifeCycleListener.latch);
+        assertClusterSizeEventually(2, h1);
+        assertClusterSizeEventually(2, h2);
+
+        Cache cacheTest = cacheManager2.getCache(cacheName);
+        assertEquals("higherHitsValue", cacheTest.get("key1"));
+        assertEquals("higherHitsValue2", cacheTest.get("key2"));
+    }
+
+    @Test
+    public void testPutIfAbsentCacheMergePolicy() {
+        String cacheName = randomMapName();
+        Config config = newConfig();
+        HazelcastInstance h1 = Hazelcast.newHazelcastInstance(config);
+        HazelcastInstance h2 = Hazelcast.newHazelcastInstance(config);
+
+        TestMemberShipListener memberShipListener = new TestMemberShipListener(1);
+        h2.getCluster().addMembershipListener(memberShipListener);
+        TestLifeCycleListener lifeCycleListener = new TestLifeCycleListener(1);
+        h2.getLifecycleService().addLifecycleListener(lifeCycleListener);
+
+        closeConnectionBetween(h1, h2);
+
+        assertOpenEventually(memberShipListener.latch);
+        assertClusterSizeEventually(1, h1);
+        assertClusterSizeEventually(1, h2);
+
+        CachingProvider cachingProvider1 = HazelcastServerCachingProvider.createCachingProvider(h1);
+        CachingProvider cachingProvider2 = HazelcastServerCachingProvider.createCachingProvider(h2);
+
+        CacheManager cacheManager1 = cachingProvider1.getCacheManager();
+        CacheManager cacheManager2 = cachingProvider2.getCacheManager();
+
+        CacheConfig cacheConfig = newCacheConfig(cacheName, PutIfAbsentCacheMergePolicy.class.getName());
+
+        Cache cache1 = cacheManager1.createCache(cacheName, cacheConfig);
+        Cache cache2 = cacheManager2.createCache(cacheName, cacheConfig);
+
+        // TODO We assume that until here and also while doing get/put, cluster is still splitted.
+        // This assumptions seems fragile due to time sensitivity.
+
+        cache1.put("key1", "PutIfAbsentValue1");
+
+        cache2.put("key1", "value");
+        cache2.put("key2", "PutIfAbsentValue2");
+
+        assertOpenEventually(lifeCycleListener.latch);
+        assertClusterSizeEventually(2, h1);
+        assertClusterSizeEventually(2, h2);
```

Comment:
Cluster size 2 doesn't means that merging operations finished, it just shows that splitted cluster finds each other but there may be running merge operations in progress. 

On the other hand, `assertOpenEventually(lifeCycleListener.latch);` shows that `MERGED` event is received and this means that merge operations are executed.